NOTES.md

```diff
@@ -0,0 +1,84 @@
+## Notes, Tips + Tricks, and General Study Guide for Interviews
+
+### Table of Contents
+- [Definitions](#definitions)
+- [Tips + Tricks](#tips--tricks)
+
+### Definitions
```

Comment:
@Puyush6889 Can you proofread these definitions and let me know if any of them do not make sense or are confusing?